nightingale_dispatcher/src/nightingale_dispatcher/mission_planner.py

```diff
@@ -114,9 +97,11 @@ def goal_cb(self, goal):
 
         while not self.phases.empty():
             phase = self.phases.get()
-            status = phase()
-
-            if not status:
+            try:
+                status = phase()
+            except Exception as e:
```

Comment:
What if we want to do some specific error handling inside the tasks though? It seems like any error raised will bring it here. Are you suggesting that we should handle all of the errors in this spot? I'm concerned if we want to handle errors, have the handling succeed in bringing the robot back but then be here and unable to go back to the state it was at